NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  TaskButton.swift
+//  NetworkingSampleApp
+//
+//  Created by Matej Molnár on 11.12.2023.
+//
+
+import SwiftUI
+
+struct TaskButton: View {
+    enum Config {
+        case play, pause, cancel, retry
+
+        var imageName: String {
+            switch self {
+            case .play: "play"
+            case .pause: "pause"
+            case .retry: "repeat"
+            case .cancel: "x"
+            }
+        }
+
+        var color: Color {
+            switch self {
+            case .play, .pause, .retry:
+                .blue
+            case .cancel: 
+                .red
+            }
+        }
+    }
+
+    let config: Config
```

Comment:
I'm always challenging the public properties even though let. Does it make sense for any other structures to have those two visible?

Reply:
True, I made them private.